src/ranged.cpp

```diff
@@ -2398,6 +2398,14 @@ void target_ui::update_target_list()
     std::sort( targets.begin(), targets.end(), [&]( const Creature * lhs, const Creature * rhs ) {
         return rl_dist_exact( lhs->pos(), you->pos() ) < rl_dist_exact( rhs->pos(), you->pos() );
     } );
+
+    // Push targets the player's attitude to them is friendly to end
+    std::stable_sort( targets.begin(), targets.end(), [&]( const Creature * lhs,
+    const Creature * rhs ) {
+        Creature::Attitude lhsAtt = you->attitude_to( *lhs );
+        Creature::Attitude rhsAtt = you->attitude_to( *rhs );
+        return lhsAtt != rhsAtt && lhsAtt != Creature::Attitude::FRIENDLY;
+    } );
```

Comment:
This doesn't need to be two separate `sort` calls; you can combine the logic into a single comparison function.

Reply:
Thanks I'll keep that in mind, though I've closed this since apparently
there's not enough difference between a neutral guard/npc and a
monster about to turn hostile, so while improving how it works near npcs
it seems like it could hurt gameplay near things that bounce between
hostile/neutral like ants and animals

Reply:
> apparently there's not enough difference between a neutral guard/npc and a monster about to turn hostile

What about `Creature::is_npc()`? Count friendly monsters & non-hostile NPCs as "friendly" and push them towards the end, while keeping non-friendly monsters and hostile NPCs as primary targets